### PR TITLE
rsx: Fix u16 index arrays overflow 

### PR DESCRIPTION
**Pull request by elad335**
- Force u32 destination index arrays when an overflow occurs while adding the vertex base index.

- Dont ignore the upper 16-bits of the restart index if u16 array is specified.

Fixes yakuza series' exploding vertices.

**before**
before
before2
**after**
after
after2

### CODE REVIEW

**Comment by raven02** [comment]
Fix https://github.com/RPCS3/rpcs3/issues/2450

**Comment by elad335** [comment]
Redone it to force u32 only when needed, using rsx::index_array_type::forced_u32. its value is UINT8_MAX so it won't affect readings from the 4-bit value of the vertex type.

**Comment by MarioSonic2987** [comment]
Fixes exploding vertices in some LEGO games.

<details>
<summary>LEGO Batman 3</summary>

image
image
</details>
<details>
<summary>The LEGO Movie</summary>

image
image

File: rpcs3/Emu/RSX/GL/GLVertexBuffers.cpp [hunk 1]
```diff
@@ -137,7 +134,7 @@ namespace
 			if (min_index >= max_index)
 			{
 				//empty set, do not draw
-				return{ 0, 0, 0, 0, std::make_tuple(get_index_type(type), offset_in_index_buffer) };
+				return{ 0, 0, 0, 0, std::make_tuple(static_cast<GLenum>(GL_UNSIGNED_INT), offset_in_index_buffer) };
```
Comment (kd-11) [inline rpcs3/Emu/RSX/GL/GLVertexBuffers.cpp]: GL_XXX are typeless macros, cast is not needed.

File: rpcs3/Emu/RSX/Capture/rsx_capture.cpp [hunk 2]
```diff
@@ -236,12 +236,12 @@ namespace rsx
 						auto fifo = vm::ptr<u16>::make(idxAddr);
 						for (u32 i = 0; i < idxCount; ++i)
 						{
-							u16 index = fifo[i];
-							if (is_primitive_restart_enabled && index == (u16)primitive_restart_index)
+							u32 index = (u32)fifo[i];
```
Comment (kd-11) [inline rpcs3/Emu/RSX/Capture/rsx_capture.cpp]: I see a lot of these and I think they are unnecessary as u16 cleanly fits into u32. Clutters the code somewhat imo.

**Pull request merged**